connection.js

```diff
@@ -375,82 +381,61 @@ class Connection {
 
         // only square boardsizes, except if all is allowed
         if (notification.width !== notification.height) {
-            if (!config.allow_all_boardsizes && !config.boardsizesranked && !config.boardsizesunranked) {
+            if (config.boardsizes && !config.boardsizesranked && !config.boardsizesunranked && !config.allow_all_boardsizes) {
                 return getBoardsizeNotSquareReject("boardsizes", notification.width, notification.height);
```

Comment:
yes!
so easy that i'll actually do it in next commit xD

Reply:
we still need the argName so i think it's convenient if we still pass it, but i added a layer to avoid repetitions :)

```Js
function getAllowedFamilyRejectResult(familyName, nameF, notif, notificationRanked) {
    const argNames = getArgNamesGRU(familyName);
    const [general, ranked, unranked] = argNames;
    const [general_underscored, ranked_underscored, unranked_underscored] = getArgNamesUnderscoredGRU(familyName);

    if (config[general] && !config[ranked] && !config[unranked] && !config[`allow_all_${general_underscored}`] && !config[`allowed_${general_underscored}`][notif]) {
        return getAllowedFamilyReject(general, nameF, notif);
    }
    if (config[ranked] && notificationRanked && !config[`allow_all_${ranked_underscored}`] && !config[`allowed_${ranked_underscored}`][notif]) {
        return getAllowedFamilyReject(ranked, nameF, notif);
    }
    if (config[unranked] && !notificationRanked && !config[`allow_all_${unranked_underscored}`] && !config[`allowed_${unranked_underscored}`][notif]) {
        return getAllowedFamilyReject(unranked, nameF, notif);
    }
}
```